src/components/chat-view/Chat.tsx

```diff
@@ -539,48 +540,57 @@ const Chat = forwardRef<ChatRef, ChatProps>((props, ref) => {
       <div className="smtcmp-chat-messages" ref={chatMessagesRef}>
         {chatMessages.map((message, index) =>
           message.role === 'user' ? (
-            <ChatUserInput
-              key={message.id}
-              ref={(ref) => registerChatUserInputRef(message.id, ref)}
-              initialSerializedEditorState={message.content}
-              onChange={(content) => {
-                setChatMessages((prevChatHistory) =>
-                  prevChatHistory.map((msg) =>
-                    msg.role === 'user' && msg.id === message.id
-                      ? {
-                          ...msg,
-                          content,
-                        }
-                      : msg,
-                  ),
-                )
-              }}
-              onSubmit={(content, useVaultSearch) => {
-                if (editorStateToPlainText(content).trim() === '') return
-                handleSubmit(
-                  [
-                    ...chatMessages.slice(0, index),
-                    {
-                      ...message,
-                      content,
-                    },
-                  ],
-                  useVaultSearch,
-                )
-                chatUserInputRefs.current.get(inputMessage.id)?.focus()
-              }}
-              onFocus={() => {
-                setFocusedMessageId(message.id)
-              }}
-              mentionables={message.mentionables}
-              setMentionables={(mentionables) => {
-                setChatMessages((prevChatHistory) =>
-                  prevChatHistory.map((msg) =>
-                    msg.id === message.id ? { ...msg, mentionables } : msg,
-                  ),
-                )
-              }}
-            />
+            <div key={message.id} className="smtcmp-chat-messages-user">
+              <ChatUserInput
+                ref={(ref) => registerChatUserInputRef(message.id, ref)}
+                initialSerializedEditorState={message.content}
+                onChange={(content) => {
+                  setChatMessages((prevChatHistory) =>
+                    prevChatHistory.map((msg) =>
+                      msg.role === 'user' && msg.id === message.id
+                        ? {
+                            ...msg,
+                            content,
+                          }
+                        : msg,
+                    ),
+                  )
+                }}
+                onSubmit={(content, useVaultSearch) => {
+                  if (editorStateToPlainText(content).trim() === '') return
+                  handleSubmit(
+                    [
+                      ...chatMessages.slice(0, index),
+                      {
+                        role: 'user',
+                        content: content,
+                        promptContent: null,
+                        id: message.id,
+                        mentionables: message.mentionables,
```

Comment:
Is there a reason this changed from `{ ...message, content }`?

Reply:
We have to reset properties like `message.similaritySearchResults` or `message.promptContent`. I think defining the entire object explicitly is clearer than using the spread syntax `{ ...message }`